### ISSUE TITLE
Collaboration opportunity?

### ISSUE THREAD

**Issue by ibgreen** [Geodan/mapbox-3dtiles#16]
Hi, very nice demo, thanks for sharing. 

We are also developing a framework-independent 3d tile loader (part of our loaders.gl effort), you can see results developing at:

- https://loaders.gl/examples/3d-tiles
- https://loaders.gl/docs/specifications/category-3d-tiles
- https://loaders.gl/docs/api-reference/3d-tiles

A few remarks:

- We created loaders.gl to enable collaboration/interoperability with others developers, such as yourselves. 
- As you have probably discovered, implementing the full 3d tiles standard is a big effort and as such, best shared. 
- If collaboration sounds interesting, happy to connect.

Also:

- Could you clarify the licenses on your open source code, ideally by adding a `LICENSE` file to your repo? As it stands, it is not clear whether we can use/copy/fork your code
- One idea would be to port your mapbox/THREE.js sample to use the `loaders.gl/3d-tiles` loader and include as a demo on loaders.gl website (with attribution to yourselves of course!)
- If we can't use your implementation as a base due to licensing issues that is OK, we'd likely roll our own, but why reinvent the wheel when working with open source?

**Comment by tomvantilburg** [comment]
Hi @ibgreen, we weren't aware of the current efforts on 3d-tiles in loaders.gl, thanks so much for sharing! We will look into it. 

In the meantime some answers: 

- As far as we are aware there is a [LICENSE](https://github.com/Geodan/mapbox-3dtiles/blob/master/LICENSE) file with a BSD-3 license in the repository.
- We are currently using the `THREE.GLTFLoader();` to load the gltf models that are inside the 3D-tiles. I assume you are also using this function from THREEjs? Do you have an example implementation of that? I only saw a pointcloud implementation in your online examples.

**Comment by ibgreen** [comment]
> As far as we are aware there is a LICENSE file with a BSD-3 license in the repository.

@tomvantilburg Yes there it is! Don't know how I could have missed that. Awesome, thanks!

**Comment by ibgreen** [comment]
> We are currently using the THREE.GLTFLoader(); to load the gltf models that are inside the 3D-tiles. I assume you are also using this function from THREEjs? 

We don't, as I will explain below, however you can take the extracted `GLB` data from the `Tile3DLoader` and pass to `THREE.GLTFLoader`.

**Comment by ibgreen** [comment]
As context, loaders.gl is a suite of **framework-independent** loaders. The idea is that they can be used with any WebGL framework.

We currently only have an integration with our own https://deck.gl framework [Tile3DLayer](https://github.com/uber/deck.gl/blob/master/modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js)

But we'd love to be able to showcase a THREE.js integration as well, and your code would be a perfect baseline to build that example on.

If you are interested in using loaders.gl as a base we could collaborate on porting your THREE.js based example to loaders.

We'd both benefit from sharing a common implementation of this quite complex standard, and we could share work on optimizing loading and traversal etc.

**Comment by ibgreen** [comment]
@tomvantilburg Quick update

- Initial port of your example to loaders.gl: https://github.com/uber-web/loaders.gl/pull/419
- Added your test tilesets to deck.gl-data: https://github.com/uber-common/deck.gl-data/pull/14

The current port uses loaders.gl to parse the tiles, but does not yet use loaders.gl traversal algorithm, see the pull request for more information.
